### ISSUE TITLE
OpenVino model generates non-deterministic output

### ISSUE THREAD

**Issue by Alkhaddour** [openvinotoolkit/openvino#13513]
##### System information (version)
- OpenVINO => 2021.4
- Operating System => Ubuntu - 20.04.4 LTS (Focal Fossa)
- Environment:
  1) conda==4.11.0
  2) python==3.8.13
  3) openvino-dev[onnx,pytorch,tensorflow2]==2021.4.2
  4) timm==0.6.7

##### Detailed description
ViT models converted from `timm` (pytorch-based) to `onnx` to `OpenVino` produce different outputs for the same input.

##### Steps to reproduce

[This code](https://github.com/Alkhaddour/ReportedBugs/blob/main/OV_2021.4.2_bug/reproduce_OV_error.py) can be used to reproduce the error, the output of running this script is [here](https://github.com/Alkhaddour/ReportedBugs/blob/main/OV_2021.4.2_bug/log.txt).

**Comment by Iffa-Meah** [comment]
@Alkhaddour I checked your custom code:


![custom](https://user-images.githubusercontent.com/79643476/196326719-a80069eb-02d8-4587-9025-5ee68a627028.png)


benchmark_app native ONNX:

![onnx](https://user-images.githubusercontent.com/79643476/196326798-96d4fa0d-32c0-4579-85e7-b421bf74731b.png)


benchmark_app IR:

![xml](https://user-images.githubusercontent.com/79643476/196326867-003401c0-fd06-4f6e-95d9-32e84bc757c8.png)


So far your files seem to be correctly generated & supported by OpenVINO.

Could you elaborate further on what you mean by "OpenVino model generates **non-deterministic** output" ?.

**Comment by Alkhaddour** [comment]
@Iffa-Meah, you are using OpenVINO-2022.1.0, the issue is raised for the previous version 2021.4.2
In the previous version, I got the following output from the OpenVINO model (using the exact same input):
```
OV output:
 [1.3498335, 0.65310645, 0.07171944, -1.4879308, -3.1347675, -4.133969, -4.4626064, -4.567269, -4.593824, -4.5874243]
```

**Comment by Iffa-Meah** [comment]
@Alkhaddour your inferencing results (Pytorch and OV) seem to be different no matter in which version.

<img width="808" alt="Untitled" src="https://user-images.githubusercontent.com/79643476/196579202-e94c3f98-bf6a-4f48-a506-b5c1c35b417f.png">

We'll further investigate this and get back to you with possible workaround.

**Comment by eaidova** [comment]
@Alkhaddour, looks like onnx model produced by torch.export is not correct, I just added running model via onnxruntime to make sure that onnx itself produces the same results as pytorch, but it gives the same behavior as openvino, the code below:

```
import onnxruntime
sess = onnxruntime.InferenceSession(onnx_path)
input_name = sess.get_inputs()[0].name

onnx_outputs = []
for i in range(10):
    res_onnx = sess.run(None, {input_name: image.astype(np.float32)})[0]
    onnx_outputs.append(res_onnx[0][0])
```

**Comment by eaidova** [comment]
@Alkhaddour I little bit confused parameters which you provide for timm. if I right remember, vit has pre-trained models only for 1000 classes in imagenet, so combination `pretrained=True` and `num_classes=1` lead to random weights initialization for last layer (please see [here](https://github.com/rwightman/pytorch-image-models/blob/475ecdfa3d369b6d482287f2467ce101ce5c276c/timm/models/helpers.py#L308)), is it expected from your side?

I tried also debug original model with 1000 classes, looks like there is some difference in onnx behaviour for logits representation, in the same time if it does not break labels scores relative comparison. I mean, if you apply np.argsort to scores, you get the same result for both pytorch and openvino. You also can try to apply softmax to convert raw logits into [0, 1].

**Comment by Apollo9999** [comment]
OnnxModel Runtime test Using Python Code 
[ONNXRUNTIME.zip](https://github.com/openvinotoolkit/openvino/files/9870830/ONNXRUNTIME.zip)

**Comment by Alkhaddour** [comment]
@eaidova, thanks for your feedback. Considering the random initialization for the last layer, it is an expected behavior. I will further investigate the problem in ONNX. I don't recall having score mismatch in ONNX, moreover, if the problem is in ONNX, then I would expect to see the same issue in all OV versions, but it is solved starting from 2022.1

**Comment by Alkhaddour** [comment]
@hbalasu1, sorry for late answer, still don't understand why the problem is solved for newer OV versions. I will further investigate the problem according to [this comment](https://github.com/openvinotoolkit/openvino/issues/13513#issuecomment-1285283732).

**Comment by Alkhaddour** [comment]
@Apollo9999, thank you, will check this too.

**Comment by hbalasu1** [comment]
Hi @Alkhaddour 
Could you let me know if your issue has been resolved?

**Comment by avitial** [comment]
Closing this, I hope previous responses were sufficient to help you proceed. Feel free to reopen and ask additional questions related to this topic.
